src/main/java/org/codehaus/mojo/exec/ExecJavaMojo.java

```diff
@@ -358,6 +371,123 @@ public void execute() throws MojoExecutionException, MojoFailureException {
+            // good enough
+            final Object[] args = Stream.of(constructor.getParameters())
+                    .map(param -> {
+                        try {
+                            return lookupParam(param);
+                        } catch (final ComponentLookupException e) {
+                            getLog().error(e.getMessage(), e);
+                            throw new IllegalStateException(e);
+                        }
+                    })
+                    .toArray(Object[]::new);
+            constructor.setAccessible(true);
+            runnable = constructor.newInstance(args);
+        } catch (final RuntimeException re) {
+            if (getLog().isDebugEnabled()) {
+                getLog().debug(
+                                "Can't inject " + runnableClass + "': " + re.getMessage() + ", will ignore injections",
+                                re);
+            }
+            final Constructor<? extends Runnable> declaredConstructor = runnableClass.getDeclaredConstructor();
+            declaredConstructor.setAccessible(true);
+            runnable = declaredConstructor.newInstance();
+        }
+        runnable.run();
+    }
+
+    private Object lookupParam(final java.lang.reflect.Parameter param) throws ComponentLookupException {
+        switch (param.getName()) {
+                // loose coupled to maven (wrapped with standard jvm types to not require it)
+            case "systemProperties": // Properties
+                return getSession().getSystemProperties();
+            case "systemPropertiesUpdater": // BiConsumer<String, String>
+                return propertiesUpdater(getSession().getSystemProperties());
+            case "userProperties": // Properties
+                return getSession().getUserProperties();
+            case "userPropertiesUpdater": // BiConsumer<String, String>
+                return propertiesUpdater(getSession().getUserProperties());
+            case "projectProperties": // Properties
+                return project.getProperties();
+            case "projectPropertiesUpdater": // BiConsumer<String, String>
+                return propertiesUpdater(project.getProperties());
+            case "highestVersionResolver": // Function<String, String>
+                return (Function<String, String>) ga -> {
+                    final int sep = ga.indexOf(':');
+                    if (sep < 0) {
+                        throw new IllegalArgumentException("Invalid groupId:artifactId argument: '" + ga + "'");
+                    }
+
+                    final org.eclipse.aether.artifact.Artifact artifact = new DefaultArtifact(ga + ":[0,)");
+                    final VersionRangeRequest rangeRequest = new VersionRangeRequest();
+                    rangeRequest.setArtifact(artifact);
+                    try {
+                        rangeRequest.setRepositories(project.getRemotePluginRepositories());
```

Comment:
why not `getRemoteProjectRepositories()`

Reply:
An error, thks
Will fix this and other uncommented issues tonight when i get back a computer

Reply:
fixed to respect the plugin configuration and use either plugin or project or both repositories